sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -564,6 +564,14 @@ def dtype(self):
     return self._expr.proxy().dtype
 
   isin = frame_base._elementwise_method('isin', base=pd.DataFrame)
+  combine_first = frame_base._elementwise_method(
+      'combine_first', base=pd.DataFrame)
+
+  combine = frame_base._proxy_method(
+      'combine',
+      base=pd.DataFrame,
+      requires_partition_by=expressions.partitionings.Singleton(),
```

Comment:
We recently added the ability for Singleton() requirements to have a "reason" specified, that describes why a particular operation is not parallelizable. The reason will be included in error messages so users have some insight. Please add one here, something like:

```suggestion
      requires_partition_by=expressions.partitionings.Singleton(
          reason="combine() is not parallelizable because func might operate on the full dataset."),
```

Reply:
FYI I filed BEAM-12351 to track making this parallelizable